### ISSUE TITLE
Set up binary releases

### ISSUE THREAD

**Issue by davidlattimore** [davidlattimore/wild#207]
Looks like the two most popular options are [cargo-dist](https://crates.io/crates/cargo-dist) and [release-plz](https://crates.io/crates/release-plz). I'm not sure what the tradeoffs between them are or which one is likely to be better for our purposes, but it's probably a good idea to have a quick look at both. @andrewdavidmackenzie, would you like to take a look?

**Comment by andrewdavidmackenzie** [comment]
Happy to take a look.
It's a bit of a pain to debug and test, creating a release each time - but there is not much strange in this build so shouldn't be too complicated.

Although working from a fork, you might need to merge before you can "test". I'll see if I can do a release on my fork though, prior to that.

**Comment by andrewdavidmackenzie** [comment]
I'll do a first version for x86, linux, gnu/musl, with a shell installer (will also work with cargo binstall)

**Comment by andrewdavidmackenzie** [comment]
By default, cargo-dist adds a "dist" profile, that inherits from release, but adds lto="thin", you can let me know if you'd like to avoid the lto....but for a release build it seems reasonable I think.

**Comment by andrewdavidmackenzie** [comment]
And
```
  × Github CI support requires you to specify the URL of your repository
  help: Set the repository = "https://github.com/..." key in these manifests:
        /home/andrew/workspace/wild/wild/Cargo.toml
        /home/andrew/workspace/wild/linker-diff/Cargo.toml
```

So I will add those too.

**Comment by andrewdavidmackenzie** [comment]
BTW: I see the wild name in crates is taken.

https://crates.io/crates/wild

**Comment by andrewdavidmackenzie** [comment]
If you would like to move a bunch of stuff to workspace Cargo.toml, and inherit them in each package (e.g. homepage, edition, license, authors, maybe align the version numbers? etc).

Then just create another issue for me...

We'd need to chose a new version number, greater than all previously, and move them all to that...

**Comment by davidlattimore** [comment]
Can we configure it to not publish anything to creates.io? We might want to at some stage if there's someone else that wants to use the linker as a library, but for now it seems like an extra thing that we don't need

**Comment by davidlattimore** [comment]
> By default, cargo-dist adds a "dist" profile, that inherits from release, but adds lto="thin", you can let me know if you'd like to avoid the lto....but for a release build it seems reasonable I think.

Thin LTO sounds good for distribution builds.

**Comment by andrewdavidmackenzie** [comment]
In fact caego-dist doesn't publish to crates.io for you.

You have to do that as a manual task from your machine.

It does have some requirement (I will need to double check the details)
that the version number in Cargo.toml matches the tag though.
Not sure what it does when there are multiple crates with different version
numbers...

Would you be OK with aligning version numbers across cratees.

**Comment by davidlattimore** [comment]
Fine with me to align version numbers. Although I got the impression that it got the version number from the git tag

**Comment by andrewdavidmackenzie** [comment]
Pushed a commit that does that to the PR.

**Comment by davidlattimore** [comment]
There's already a 0.2.0 tag... from March. Must be time for a new version :) But I'll do that tomorrow
